### ISSUE TITLE
Enhancement - Web forms for Campaigns and Support Service Cases

### ISSUE THREAD

**Issue by c-nv-s** [cortezaproject/corteza-webapp-one#53]
As a user I would like to be able to create web forms that I can embed on an external site.
the web forms can be used either for:
- lead acquisition ( on completion a new lead is created )
- support case creation ( on completion a new support case is created )

The value added would be that it would enable the automation of the creation of leads and support cases via websites.

**Describe the solution you'd like**
Perhaps an area in the CRM or Support Service apps which allow a user to create a web form.
The html/javascript generated could be copied by a user and embedded onto a company website or support page.
when a visitor to that company website completes and submits the web form, the form sends the details via the corteza-server api and it is stored in the system.

**Describe alternatives you've considered**
At the moment I believe you would have to use the feature from a separate application such as Mautic, Mailchimp etc. and then export the leads from there and try to import them into Corteza.
I suppose you could develop an "import from... " feature but I think this functionality can usefully live as part of Corteza itself.

**Comment by zmija** [comment]
Hi @c-nv-s,

Thank you for submitting this, I like the idea, and I see the value it can bring. In fact, some of our customers already use it in such a way, so Corteza is collecting the data from forms submitted on WordPress pages or standalone web applications via API.

I'll discuss a time effort with our devs for the solution you outlined, but this is definitely something to seriously consider for the near future.

In the meantime - can we help you with some instructions on how to achieve that using API calls? We also have experimental integration with Mautic in place, if you are interested? If you give us more concrete information about what you wish to achieve, we can find a way to hack it.

Cheers!

**Comment by c-nv-s** [comment]
I would definitely appreciate if you could point me to any existing resources/instructions such as guides, plugins/integrations for all three: wordpress, mautic and api.
I am very keen on giving the all-in-one docker-compose repo a try to test some of these things however I'm not sure it's ready for testing yet?

**Comment by c-nv-s** [comment]
@zmija did you have any links?

**Comment by tjerman** [comment]
@c-nv-s Currently I can only paste you an example of how we send contacts from Mautic to Corteza based on our use-case -- https://gist.github.com/tjerman/785a775936b1dc747360c250c357ba39
Here are the docs that go more into details on how to do custom stuff with Corteza -- https://docs.cortezaproject.org/corteza-docs/2020.12/integrator-guide/index.html
These resources should enable you to do pretty much anything.

Basically; anything that knows how to send webhooks (Mautic, WordPress, ...) can be integrated with Corteza following the above example script.

We are planning on implementing proper plug-and-play integrations for different services in the future releases (I don't have a timeline for that yet).

**Comment by c-nv-s** [comment]
thanks, I will make use of these resources
